### PR TITLE
Integrate with modular Iam API

### PR DESCRIPTION
**Pull request by JGiter**
- Methods from iam-lib are spread over modules
- session management is moved from iam-lib to loginService, as it is specific to login logic

### CODE REVIEW

**Comment by JGiter** [comment]
> Please do not change code 

Ok, sorry. By the way, don't you configured prettier? It would ensure consistent code style

File: src/app/shared/services/login/login.service.ts [hunk 1]
```diff
@@ -75,9 +92,12 @@ export class LoginService {
    * Disconnect from IAM
    */
   logout(saveDeepLink?: boolean) {
+    this.clearSession();
     this.iamService.closeConnection().subscribe(() => {
       saveDeepLink ? this.saveDeepLink() : location.href = location.origin + '/welcome';
-    });
+    });;
```
Comment (dawidgil) [inline src/app/shared/services/login/login.service.ts:98]: double semicolon

File: src/app/shared/services/login/login.service.ts [hunk 2]
```diff
@@ -75,9 +92,12 @@ export class LoginService {
...
+
+    saveDeepLink ? this.saveDeepLink() : location.href = location.origin + '/welcome';
```
Comment (dawidgil) [inline src/app/shared/services/login/login.service.ts:100]: why adding this?

File: src/app/state/auth/auth.effects.ts [hunk 3]
```diff
@@ -134,16 +136,17 @@ export class AuthEffects {
   reinitializeLoggedUser$ = createEffect(() =>
     this.actions$.pipe(
       ofType(AuthActions.reinitializeAuth, AuthActions.reinitializeAuthForPatron, AuthActions.reinitializeAuthForEnrol),
-      filter(() => this.loginService.isSessionActive()),
+      filter(this.loginService.isSessionActive),
+      // filter(() => await this.loginService.isSessionActive()),
```
Comment (dawidgil) [inline src/app/state/auth/auth.effects.ts:140]: please remove this comment.

File: src/app/state/auth/auth.effects.ts [hunk 4]
```diff
@@ -162,17 +165,17 @@ export class AuthEffects {
   notPossibleToReinitializeUser$ = createEffect(() =>
     this.actions$.pipe(
       ofType(AuthActions.reinitializeAuth),
-      filter(() => !this.loginService.isSessionActive()),
+      filterAsync(async (e) => !this.loginService.isSessionActive()),
```
Comment (dawidgil) [inline src/app/state/auth/auth.effects.ts:168]: is this filterAsync still needed?

File: src/app/routes/registration/request-claim/request-claim.component.ts [hunk 5]
```diff
@@ -272,10 +273,10 @@ export class RequestClaimComponent implements OnInit, SubjectElements {
   goToEnrolment() {
     if (this.roleTypeForm.value.enrolFor === EnrolForType.ASSET) {
       // Navigate to My Enrolments Page
-      this.route.navigate(['dashboard'], {queryParams: {returnUrl: '/assets/enrolment/' + this.roleTypeForm.value.assetDid}});
+      this.route.navigate(['dashboard'], { queryParams: { returnUrl: '/assets/enrolment/' + this.roleTypeForm.value.assetDid } });
```
[the remark this answers is not in the document]
Comment (JGiter) [reply src/app/routes/registration/request-claim/request-claim.component.ts:276]: Would you mind if I will add prettier?

**Review by dawidgil — APPROVED** [review]

**Pull request merged**